ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/ContractKeysIT.scala

```diff
@@ -0,0 +1,465 @@
+              RecordField(value = Some(Value(Value.Sum.Text(key)))),
+              RecordField(value = expected match {
+                case None => Value(Value.Sum.Optional(Optional(None)))
+                case Some(cid) => Value(Value.Sum.Optional(Optional(Some(cid.asContractId))))
+              }),
+            )
+        )
+        fetchResult <- ctx.testingHelpers.failingExercise(
+          testIdsGenerator.testCommandId("TDVl5"),
+          submitter = delegate,
+          template = templateIds.delegation,
+          contractId = delegationEv.contractId,
+          choice = "FetchDelegated",
+          arg = Value(Value.Sum.Record(fetchArg)),
+          Code.INVALID_ARGUMENT,
+          pattern = "dependency error: couldn't find contract"
+        )
+        // this fetch still fails even if we do not check that the submitter
+        // is in the lookup maintainer, since we have the visibility check
+        // implement as part of #753.
+        _ <- if (languageVersion precedes LanguageVersion.checkSubmitterInMaintainers) {
+          ctx.testingHelpers.failingExercise(
+            testIdsGenerator.testCommandId("TDVl6"),
+            submitter = delegate,
+            template = templateIds.delegation,
+            contractId = delegationEv.contractId,
+            choice = "FetchByKeyDelegated",
+            arg = Value(Value.Sum.Record(lookupArg(None))),
+            Code.INVALID_ARGUMENT,
+            "couldn't find key",
+          )
+        } else {
+          ctx.testingHelpers.failingExercise(
+            testIdsGenerator.testCommandId("TDVl6"),
+            submitter = delegate,
+            template = templateIds.delegation,
+            contractId = delegationEv.contractId,
+            choice = "FetchByKeyDelegated",
+            arg = Value(Value.Sum.Record(lookupArg(None))),
+            Code.INVALID_ARGUMENT,
+            "Expected the submitter 'Charlie' to be in maintainers 'Bob'",
+          )
+        }
+        _ <- if (languageVersion precedes LanguageVersion.checkSubmitterInMaintainers) {
+          ctx.testingHelpers.simpleExercise(
+            testIdsGenerator.testCommandId("TDVl7"),
+            submitter = delegate,
+            template = templateIds.delegation,
+            contractId = delegationEv.contractId,
+            choice = "LookupByKeyDelegated",
+            arg = Value(Value.Sum.Record(lookupArg(None))),
+          )
+        } else {
+          ctx.testingHelpers.failingExercise(
+            testIdsGenerator.testCommandId("TDVl7"),
+            submitter = delegate,
+            template = templateIds.delegation,
+            contractId = delegationEv.contractId,
+            choice = "LookupByKeyDelegated",
+            arg = Value(Value.Sum.Record(lookupArg(None))),
+            Code.INVALID_ARGUMENT,
+            "Expected the submitter 'Charlie' to be in maintainers 'Bob'",
+          )
+        }
+      } yield (succeed)
+    }
+
+    //      // this is basically a port of
```

Comment:
Minor: double comment token. :slightly_smiling_face:

Reply:
Addressed in 9bf50c0.